src/nupic/frameworks/opf/metrics.py

```diff
@@ -245,119 +319,116 @@ def _isNumber(value):
 
 class MetricsIface(object):
   """
-  A Metrics module compares a prediction Y to corresponding ground truth X and returns a single
-  measure representing the "goodness" of the prediction. It is up to the implementation to
-  determine how this comparison is made.
+  A Metrics module compares a prediction Y to corresponding ground truth X and 
+  returns a single measure representing the "goodness" of the prediction. It is 
+  up to the implementation to determine how this comparison is made.
 
+  :param metricSpec: (:class:`MetricSpec`) spec used to created the metric
   """
 
   __metaclass__ = ABCMeta
 
   @abstractmethod
   def __init__(self, metricSpec):
-    """
-        instantiate a MetricsIface-based module.
-        args:
-            metricSpec is an instance of MetricSpec
-
-    """
+    pass
 
   @abstractmethod
   def addInstance(self, groundTruth, prediction, record = None, result = None):
-    """ add one instance consisting of ground truth and a prediction.
-
-      Parameters:
-      -----------------------------------------------------------------------
-      groundTruth:
-        The actual measured value at the current timestep
-      prediction:
-        The value predicted by the network at the current timestep
-
-      groundTruthEncoding:
-        The binary encoding of the groundTruth value (as a numpy array). Right
-        now this is only used by CLA networks
-      predictionEncoding:
-        The binary encoding of the prediction value (as a numpy array). Right
-        now this is only used by CLA networks
-
-      result:
-        An ModelResult class (see opf_utils.py)
-
-        return:
-            The average error as computed over the metric's window size
+    """ 
+    Add one instance consisting of ground truth and a prediction.
+
+    :param groundTruth:
+      The actual measured value at the current timestep
+    
+    :param prediction:
+      The value predicted by the network at the current timestep
+
+    :param record: ?
```

Comment:
This is the input record as fed to the `Model.compute` method by the user. The typical usage is to feed a record to that method and get a `ModelResult`. Then you pass `ModelResult.rawInput` into this function as the `record` parameter.